src/deroff.rs

```diff
@@ -538,6 +539,31 @@ impl Deroffer {
         unimplemented!()
     }
 
+    /// `condputs` (cond)itionally (puts) `s` into `self.output`
+    /// if `self.tr` is set, instead of putting `s` into `self.output` directly, 
+    /// it `translate`s it using the set translation table and puts the result
+    /// into `self.output`
+    fn condputs(&mut self, s: &str) {
+        let is_special = {
+            self.pic     ||
+            self.eqn     ||
+            self.refer   ||
+            self.r#macro ||
+            self.inlist  ||
+            self.inheader
+        };
+
+        if !is_special {
+            if let Some(table) = self.tr.clone() {
```

Comment:
Why is the `.clone()` needed here instead of having it below?

Reply:
'HashMap` isn't `Copy` which is required for `if let Some(table)` so it moves `table` out of `self.tr`, which means that `self.tr` is no longer valid, so we have to `clone` so we don't lose the translation table

(If `HashMap` was `Copy` we wouldn't need any `clones` at all)

Reply:
To avoid having to clone, `translate` should take `&HashMap<_, _>` instead of `HashMap<_, _>`.